agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/StatsbeatModule.java

```diff
@@ -46,51 +46,85 @@
   private final NetworkStatsbeat networkStatsbeat;
   private final AttachStatsbeat attachStatsbeat;
   private final FeatureStatsbeat featureStatsbeat;
+  private final FeatureStatsbeat instrumentationStatsbeat;
 
   private final AtomicBoolean started = new AtomicBoolean();
+  private final AtomicBoolean disabledAll = new AtomicBoolean();
```

Comment:
can change this to volatile boolean (not using atomic-ness)

Reply:
why not? what about started? we use atomicboolean in many places. are you concerning that too many atomic operations will add perf overhead?

Reply:
started is using it's atomic-ness: `started.getAndSet(true)`

Reply:
set and getAndSet are both atomic-ness, correct? both use Varhandle.setVolatile behind the scene.. getAndSet will return the previous value, set doesn't. not sure other differences between them... i probably don't get your point :(